File: 04-TennisScoring/java/src/test/java/TennisGameTest.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;
import org.junit.runners.Parameterized.Parameters;

import java.util.Collection;

@RunWith(Parameterized.class)
public class TennisGameTest {

  private int player1Score;
  private int player2Score;
  private String expectedScore;

  @Parameters(name = " {0}-{1} is {2} ")
  public static Collection<Object[]> getAllScores() {
    return TestData.getTestCases();
  }

  public TennisGameTest(int player1Score, int player2Score, String expectedScore) {
    this.player1Score = player1Score;
    this.player2Score = player2Score;
    this.expectedScore = expectedScore;
  }

  @Test
  public void scoreShouldBeCorrect() {
    TennisGame game = new TennisGame("player1", "player2");

    scorePoints(game, "player1", this.player1Score);
    scorePoints(game, "player2", this.player2Score);

    assertEquals(this.expectedScore, game.getScore());
  }

  private void scorePoints(TennisGame game, String player, int points) {
    for (int i = 0; i < points; i++) {
      game.wonPoint(player);
    }
  }

}

File: 04-TennisScoring/csharp-dotnetframework/TennisScoringTest/TennisGameTest.cs
using System;
using NUnit.Framework;

namespace Katas
{
    [TestFixture]
    public class TennisGameTest
    {

        [Test, TestCaseSource(typeof(TennisTestData), nameof(TennisTestData.TestData))]
        public void ScoreShouldBeDisplayedCorrectly(int p1Score, int p2Score, string expectedScore)
        {
            var game = new TennisGame("player1","player2");

            ScorePoints(game, "player1", p1Score);
            ScorePoints(game, "player2", p2Score);

            var actualScore = game.GetScore();
            Assert.AreEqual(expectedScore, actualScore);
        }

        private static void ScorePoints(ITennisGame game, string playerName, int points)
        {
            for (var i = 0; i < points; i++)
            {
                game.WonPoint(playerName);
            }
        }
    }
}
